auth-api/src/auth_api/services/affiliation.py

```diff
@@ -350,17 +352,39 @@ def fix_stale_affiliations(org_id: int, entity_details: Dict, **kwargs):
         current_app.logger.debug('>fix_stale_affiliations')
 
     @staticmethod
-    def _get_nr_details(nr_number: str, token: str):
-        """Return NR details by calling legal-api."""
-        legal_api_url = current_app.config.get('LEGAL_API_URL') + current_app.config.get('LEGAL_API_VERSION_2')
-        get_nr_url = f'{ legal_api_url }/nameRequests/{nr_number}'
-        try:
-            get_nr_response = RestService.get(get_nr_url, token=token)
-        except (HTTPError, ServiceUnavailableException) as e:
-            current_app.logger.info(e)
-            raise BusinessException(Error.DATA_NOT_FOUND, None) from e
-
-        return get_nr_response.json()
+    async def _get_affiliation_details(affiliations: List[AffiliationModel]):
+        """Return affiliation details by calling the source api."""
+        call_info = {}  # i.e. turns into { url: [identifiers...] }
+        for affiliation in affiliations:
+            url = affiliation.affiliation_details_url
+            if url in call_info:
+                call_info[url].append(affiliation.entity.business_identifier)
+            else:
+                call_info[url] = [affiliation.entity.business_identifier]
```

Comment:
`call_info[url] =  [affiliation.entity.business_identifier for affiliation in affiliations]`

Reply:
`call_info.setdefault(affiliation.entity.business_identifier, []).append(url)`

Reply:
nice I didn't know this was a thing

Reply:
me neither, but I had a gut feeling there was a way to one line this